docs/api/errors.rst

```diff
@@ -50,12 +50,16 @@ These classes are exposed both by this module and the root `psycopg` module.
 
     .. autoattribute:: pgconn
 
-        Most likely it will be in `~psycopg.pq.ConnStatus.BAD` state;
+        It has been closed and will be in `~psycopg.pq.ConnStatus.BAD` state;
         however it might be useful to verify precisely what went wrong, for
         instance checking the `~psycopg.pq.PGconn.needs_password` and
         `~psycopg.pq.PGconn.used_password` attributes.
+        Attempting to operate this connection will raise an
+        :exc:`OperationalError`.
 
         .. versionadded:: 3.1
+        .. versionchanged:: 3.1.10
+            The connection is closed before setting the attribute.
```

Comment:
I am not sure this is a behavioural change we need to advertise. Whether the connection was open, but broken, or closed, it doesn't change the end users (except for fixing the issue reported by the OP about the resource leak, for which the note in the news file is usually enough).